### PR TITLE
Disallow xml interpolation for xmlns attributes

### PR DESCRIPTION
**Pull request by rdhananjaya**
## Purpose
Disallow XML attribute interpolation for namespace attributes.

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/18938

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java [hunk 1]
```diff
@@ -2363,6 +2365,13 @@ public void visit(BLangXMLElementLiteral bLangXMLElementLiteral) {
         resultType = types.checkType(bLangXMLElementLiteral, symTable.xmlType, expType);
     }
 
+    private boolean isXmlNamespaceAttribute(BLangXMLAttribute attribute) {
+        BLangXMLQName attrName = (BLangXMLQName) attribute.name;
+        return (attrName.prefix.value.equals("")
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java]: Shall we use `.isEmpty()` instead?

File: compiler/ballerina-lang/src/main/resources/compiler.properties [hunk 2]
```diff
@@ -473,6 +473,9 @@ error.invalid.namespace.declaration=\
 error.cannot.update.xml.sequence=\
   cannot update an xml sequence
 
+error.invalid.xml.ns.interpolation=\
+  invalid xml namespace interpolation
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/resources/compiler.properties]:
```suggestion
  xml namespaces cannot be interpolated
```

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/xml/XMLLiteralTest.java [hunk 3]
```diff
@@ -108,6 +108,12 @@ public void testXMLNegativeSemantics() {
         // XML elements with mismatching start and end tags
         BAssertUtil.validateError(negativeResult, index++, "mismatching start and end tags found in xml element",
                                   73, 18);
+        // XML interpolation is not allowed to interpolate XML namespace attributes
+        BAssertUtil.validateError(negativeResult, index++, "invalid xml namespace interpolation",
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/xml/XMLLiteralTest.java]: Can merge with L113? Same for L114 and L115

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java [hunk 4]
```diff
@@ -2332,17 +2332,19 @@ public void visit(BLangXMLElementLiteral bLangXMLElementLiteral) {
...
-        });
+        }
 
         // Visit attributes.
         bLangXMLElementLiteral.attributes.forEach(attribute -> {
-            if (attribute.name.getKind() != NodeKind.XML_QNAME
-                    || !((BLangXMLQName) attribute.name).prefix.value.equals(XMLConstants.XMLNS_ATTRIBUTE)) {
+            if (attribute.name.getKind() != NodeKind.XML_QNAME || !isXmlNamespaceAttribute(attribute)) {
                 checkExpr(attribute, xmlElementEnv, symTable.noType);
             }
```
Comment (gimantha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java]: is it not possible to add this to the 'else' block of the if block at line no. 2336 and remove this?

Comment (rdhananjaya) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java]: It turned out that first iteration defines the xml namespaces, and the second iteration depends on those previously defined namespaces, there for it's not possible to merge the two into single foreach statement.

**Pull request merged**